File: docs-examples/example-kotlin-ksp/src/main/kotlin/io/micronaut/configuration/openapi/docs/controllers/HelloController.kt
package io.micronaut.configuration.openapi.docs.controllers

// tag::imports[]
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import reactor.core.publisher.Mono

// end::imports[]
// tag::clazz[]
@Controller("/")
open class HelloController {

    /**
     * @param name The person's name
     * @return The greeting
     */
    @Get(uri = "/hello/{name}", produces = [MediaType.TEXT_PLAIN])
    open fun index(name: String): Mono<String> {
        return Mono.just("Hello $name!")
    }
}
// end::clazz[]

File: docs-examples/example-python/src/main/python/micronaut/configuration/openapi/docs/controllers/HelloController.py
# tag::imports[]
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get
from reactor.core.publisher import Mono
# end::imports[]
# tag::clazz[]
@Controller
class HelloController:

    @Get(uri="/hello/{name}", produces=MediaType.TEXT_PLAIN)
    def index(self, name: str) -> Mono[str]:
        """
        @param name The person's name
        @return The greeting
        """
        return Mono.just("Hello " + name + "!")
# end::clazz[]
